src/QMCWaveFunctions/LCAO/MultiFunctorAdapter.h

```diff
@@ -57,6 +59,37 @@ struct MultiFunctorAdapter
       u[i] = Rnl[i]->f(r);
   }
 
+  inline void batched_evaluate(OffloadArray2D& r, OffloadArray3D& u, RealType Rmax) const
+  {
+    r.updateFrom(); // TODO: remove after offload
+    const size_t nElec = r.size(0);
+    const size_t Nxyz  = r.size(1); // number of PBC images
+    assert(nElec == u.size(0));
+    assert(Nxyz == u.size(1));
+    const size_t nRnl = u.size(2);    // number of splines
+    const size_t nR   = nElec * Nxyz; // total number of positions to evaluate
+
+    auto* r_ptr = r.data();
+    auto* u_ptr = u.data();
+
+
+    for (size_t ir = 0; ir < nR; ir++)
+    {
+      if (r_ptr[ir] >= Rmax)
+      {
+        for (size_t i = 0, n = Rnl.size(); i < n; ++i)
+          u_ptr[ir * nRnl + i] = 0.0;
+      }
+      else
+      {
+        for (size_t i = 0, n = Rnl.size(); i < n; ++i)
+          u_ptr[ir * nRnl + i] = Rnl[i]->f(r_ptr[ir]);
+      }
+    }
```

Comment:
More of my taste. Remove all `{}` on lines 77-88.

Reply:
fixed